src/site/resources/styleguides/google-java-style-20180523/javaguide.html

```diff
@@ -1102,23 +1151,24 @@ <h4 id="s7.1.1-javadoc-multi-line">7.1.1 General form</h4>
 </pre>
 
 <p>The basic form is always acceptable. The single-line form may be substituted when the entirety
-of the Javadoc block (including comment markers) can fit on a single-line. Note that this only
+of the Javadoc block (including comment markers) can fit on a single line. Note that this only
 applies when there are no block tags such as <code>@return</code>.
 
 </p><h4 id="s7.1.2-javadoc-paragraphs">7.1.2 Paragraphs</h4>
 
-<p>One blank line&#8212;that is, a line containing only the aligned leading asterisk
-(<code>*</code>)&#8212;appears between paragraphs, and before the group of block tags if
-present. Each paragraph but the first has <code>&lt;p&gt;</code> immediately before the first word,
-with no space after.</p>
+<p>One blank line—that is, a line containing only the aligned leading asterisk
+(<code>*</code>)—appears between paragraphs, and before the group of block tags if present.
+Each paragraph except the first has <code>&lt;p&gt;</code> immediately before the first word, with
+no space after it. HTML tags for other block-level elements, such as <code>&lt;ul&gt;</code> or
+<code>&lt;table&gt;</code>, are <em>not</em> preceded with <code>&lt;p&gt;</code>.</p>
```

Comment:
https://www.w3schools.com/html/html_blocks.asp

>  ```<address>
<article>
<aside>
<blockquote>
<canvas>
<dd>
<div>
<dl>
<dt>
<fieldset>
<figcaption>
<figure>
<footer>
<form>
<h1>-<h6>
<header>
<hr>
<li>
<main>
<nav>
<noscript>
<ol>
<p>
<pre>
<section>
<table>
<tfoot>
<ul>
<video>
```

looks like list of tags to address is a bit bigger.

Reply:
@Zopsss, lets create separate issue on this and share details on how different actual html page will looks like if `<p>` is missed. I loosing a reason why `p` is not required in example above.

Reply:
if we can not find a reason ourself, we can ask `[cushon](https://github.com/google/styleguide/commits?author=cushon)` at https://github.com/google/styleguide/commit/49488412b2f59843fce2433bd834a3fd9700604e#diff-10efc97bb07838e0dcb3978a42e46089cca7c09cd37aa2082efc93db2b27ab4fR1162

Reply:
> @Zopsss, lets create separate issue on this and share details on how different actual html page will looks like if `<p>` is missed. I loosing a reason why `p` is not required in example above.

<p> itself is a block level element, which can be used to start new block ( i.e show information on new line ). 

While other block tags are automatically displayed on new lines so we don't need preceding <p> for them. But I'm just guessing this, I'm not sure if this is the correct reason or not.

Reply:
We are not here to guess, let's test it to be sure. If we see problem please ask Google engineer about clarification on this rule.

Reply:
Ok, we have issue https://github.com/checkstyle/checkstyle/issues/15011

We can update coverage table with comment that there's know issue to cover it fully, and link to chapter should be updated to new cached website.
And we can close this diff discussion until that issue.

Reply:
@Zopsss , fact that there is no special token is not definition that it is impossible.

We have:
```
|--HTML_ELEMENT_START -> HTML_ELEMENT_START [4:3]
|       |   |--START -> < [4:3]
|       |   |--HTML_TAG_NAME -> h1 [4:4]
|       |   `--END -> > [4:6]
```

It will be not convenient and easy - yes. But it is possible and doable.

Reply:
> Ok, we have issue #15011
> 
> We can update coverage table with comment that there's know issue to cover it fully, and link to chapter should be updated to new cached website. And we can close this diff discussion until that issue.

done https://github.com/checkstyle/checkstyle/pull/15022

Reply:
#15011 is placed in todo queue of project.

Reply:
coverage is updated at https://github.com/checkstyle/checkstyle/pull/15022/files

review thread is resolved.